projects/igniteui-angular/src/lib/grids/tree-grid/tree-grid.component.ts

```diff
@@ -601,6 +601,13 @@ export class IgxTreeGridComponent extends IgxGridBaseComponent implements IGridD
     public addRow(data: any, parentRowID?: any) {
         if (parentRowID) {
             super.endEdit(true);
+
+            const state = this.transactions.getState(parentRowID);
+            // we should not allow adding of rows as child of deleted row
+            if (state && state.type === TransactionType.DELETE) {
+                return;
```

Comment:
Shouldn't we throw error instead of failing silently?

Reply:
Yeah, will add this.